packages/app/background-jobs-common/src/background-job-processor/processors/AbstractBackgroundJobProcessor.ts

```diff
@@ -215,6 +141,7 @@ export abstract class AbstractBackgroundJobProcessor<
   }
 
   private registerListeners() {
+    // TODO: extract hooks handling to a separate class
```

Comment:
To not delay this more and not have an even bigger PR, as hooks refactor will not imply a breaking change, leaving it for a follow-up PR